File: src/main/kotlin/data/NendoroidSet.kt
package data

import basePath
import java.io.File
import java.nio.file.Paths

data class NendoroidSet(
    override var num: String = "",
    val setName: String = "",
    val list: MutableList<String> = mutableListOf()
): DataWritable {

    override fun find(): File {
        val path = Paths.get(basePath, "Set")
        val number = String.format("%03d", num.toIntOrNull() ?: -1)
        return File(path.toFile(), "Set$number.json")
    }

    override fun toString(): String {
        return "$setName : $list"
    }
}

File: GSCParser/Model/NendoroidSet.swift
//
//  NendoroidSet.swift
//  Parser
//
//  Created by 이승윤 on 2022/12/02.
//

public struct NendoroidSet: Base {
    
    public func location() -> String {
        return "Set/\(String(format: "Set%03d", num.toInt()!)).json"
    }

    public let num: String
    public var setName: String
    public var list: [String]
    
    public init(num: String, setName: String, list: [String]) {
        self.num = num
        self.setName = setName
        self.list = list
    }
}
